src/screens/AddressDiscoveryScreen.tsx

```diff
@@ -67,6 +68,9 @@ const AddressDiscoveryScreen = ({ navigation, route: { params }, ...props }: Scr
 
   const stopScan = () => dispatch(addressDiscoveryStopped())
 
+  const continueToNextScreen = () =>
+    resetNavigationState(isImporting ? 'NewWalletSuccessScreen' : 'InWalletTabsNavigation')
```

Comment:
Resetting the navigation state helps with preventing going back to the wallet creation procedure. There's no point doing that since the wallet has already been created at this point.